debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java

```diff
@@ -846,6 +860,72 @@ int allSlotsInVerificationTypeInfoDo(U8Pointer cursor) throws CorruptDataExcepti
 		}
 		return 3;
 	}
+
+	void recordAttributeDo(U32Pointer attribute) throws CorruptDataException
+	{
+		if (attribute.isNull()) {
+			return;
+		}
+		U32Pointer attributeStart = attribute;
+		int numRecordComponents = attribute.at(0).intValue();
+		classWalkerCallback.addSlot(clazz, SlotType.J9_U32, attribute, "numberRecordComponents");
+		attribute = attribute.add(1);
+		for (int i = 0; i < numRecordComponents; i++) {
+			J9ROMRecordComponentShapePointer recordComponent = J9ROMRecordComponentShapePointer.cast(attribute);
+			attribute = U32Pointer.cast(recordComponent.add(1));
+
+			J9ROMNameAndSignaturePointer recordComponentNAS = recordComponent.nameAndSignature();
+			classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, recordComponentNAS.nameEA(), "name");
+			classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, recordComponentNAS.signatureEA(), "signature");
+			classWalkerCallback.addSlot(clazz, SlotType.J9_U32, recordComponent.attributeFlagsEA(), "attributeFlags");
+			classWalkerCallback.addSection(clazz, recordComponent, attribute.getAddress() - recordComponent.getAddress(), "recordComponentShape", true);
```

Comment:
The length could be expressed more clearly as `J9ROMRecordComponentShape.SIZEOF`.

Reply:
updated with this change.